qadence2_platforms/backends/fresnel1/register.py

```diff
@@ -2,35 +2,71 @@
 
 import warnings
 
-import numpy as np
-from pulser.devices._devices import AnalogDevice
 from pulser.register.register_layout import RegisterLayout
 from qadence2_ir.types import Model
 
+from .device_settings import Fresnel1Settings
+from .._base_analog.register import RegisterTransform, RegisterResolver
+
 warnings.filterwarnings("ignore", category=UserWarning)
 warnings.formatwarning = lambda msg, *args, **kwargs: f"WARNING: {msg}\n"
 
 
 def from_model(model: Model) -> RegisterLayout:
-    model_register = model.register
+    register = RegisterResolver.resolve_from_model(
+        model=model,
+        device_settings=Fresnel1Settings,
+        grid_scale_fn=check_grid_scale,
+        grid_type_fn=check_grid_type,
+        directives_fn=check_directives,
+        register_transform_fn=from_layout,
+    )
+
+    return register
+
 
-    if model_register.num_qubits < 1:
-        raise ValueError("No qubit available in the register.")
+def from_layout(register_transform: RegisterTransform) -> RegisterLayout:
+    """
+    Transforms a data from coordinates into a register with calibrated layout.
 
-    if model_register.grid_type and model_register.grid_type != "triangular":
+    Args:
+        register_transform (RegisterTransform): the instance of RegisterTransform containing
+        the coordinates and device settings
+
+    Returns:
+        A register data with calibrated layout
+    """
+
+    layout = register_transform.get_calibrated_layout("TriangularLatticeLayout(61, 5.0µm)")
+    traps = layout.get_traps_from_coordinates(*(register_transform.coords * 5))
+    register = layout.define_register(*traps, qubit_ids=range(len(traps)))
```

Comment:
This magic multiplication by 5 I think is effectively defining the length dimensions for coordinates coming from the expressions. Probably for now it needs to go the `_base_analog`, or else we get different dimensions in different backends (currently the `analog` backend I think does not apply any scaling and thus will fail if the qubits are too close).

Reply:
But either way, we will need a better way to centralize unit conversions.

Reply:
it's now on the issue #38